src/comms.cpp

```diff
@@ -20,67 +21,272 @@
+byte secplus1States[] = {0x35,0x35,0x33,0x33, 0x38,0x3A,0x39};
 
 /*************************** FORWARD DECLARATIONS ******************************/
 
 void sync();
-bool transmit(PacketAction& pkt_ac);
+bool process_PacketAction(PacketAction& pkt_ac);
 void door_command(DoorAction action);
 void send_get_status();
+bool transmitSec1(byte toSend);
+bool transmitSec2(PacketAction& pkt_ac);
 
 /********************************** MAIN LOOP CODE *****************************************/
 
 void setup_comms() {
-    RINFO("Setting up comms for secplus2.0 protocol");
 
-    sw_serial.begin(9600, SWSERIAL_8N1, UART_RX_PIN, UART_TX_PIN, true);
-    sw_serial.enableIntTx(false);
-    sw_serial.enableAutoBaud(true); // found in ratgdo/espsoftwareserial branch autobaud
+    // init queue
+    q_init(&pkt_q, sizeof(PacketAction), 8, FIFO,  false);
 
     LittleFS.begin();
-    id_code = read_file_from_flash("id_code");
-    if (!id_code) {
-        RINFO("id code not found");
-        id_code = (random(0x1, 0xFFF) << 12) | 0x539;
-        write_file_to_flash("id_code", &id_code);
-    }
-    RINFO("id code %02X", id_code);
 
-    rolling_code = read_file_from_flash("rolling");
-    RINFO("rolling code %02X", rolling_code);
+    gdoSecurityType = read_gdo_security_from_flash("gdo_security");
 
-    q_init(&pkt_q, sizeof(PacketAction), 5, FIFO,  false);
+    if (gdoSecurityType == 1) {
 
-    RINFO("Syncing rolling code counter after reboot...");
-    sync();
+        RINFO("Setting up comms for Secuirty+1.0 protocol");
 
+        sw_serial.begin(1200, SWSERIAL_8E1, UART_RX_PIN, UART_TX_PIN, true);
+
+        wallPanelDetected = false;
+        wallplateBooting = false;
+        doorState = DoorState::Unknown;
+        lightState = 2;
+        lockState  = 2;
+
+    }
+    else {
+        RINFO("Setting up comms for Secuirty+2.0 protocol");
+
+        sw_serial.begin(9600, SWSERIAL_8N1, UART_RX_PIN, UART_TX_PIN, true);
+        sw_serial.enableIntTx(false);
+        sw_serial.enableAutoBaud(true); // found in ratgdo/espsoftwareserial branch autobaud
+
+        id_code = read_file_from_flash("id_code");
+        if (!id_code) {
+            RINFO("id code not found");
+            id_code = (random(0x1, 0xFFF) << 12) | 0x539;
+            write_file_to_flash("id_code", &id_code);
+        }
+        RINFO("id code %02X", id_code);
+
+        rolling_code = read_file_from_flash("rolling");
+        RINFO("rolling code %02X", rolling_code);
+
+        RINFO("Syncing rolling code counter after reboot...");
+        sync();
+    }
+}
+
+void wallPlate_Emulation() {
+    
+	if (wallPanelDetected) return;
+
+	unsigned long currentMillis = millis();
+	static unsigned long lastRequestMillis = 0;
+	static bool emulateWallPanel = false;
+	static unsigned long serialDetected = 0;
+	static uint8_t stateIndex = 0;
+
+	if (!serialDetected) {
+		if (sw_serial.available()) {
+			serialDetected = currentMillis;
+		}
+
+		return;
+	}
+
+    // 
```

Comment:
picky, but there are several formatting cleanups.  remove this empty comment, and fixup the mixed indentations below

Reply:
FYI that I use Visual Studio Code to edit, and it has a C/C++ extension which will format your code (shift-opt-F on Apple keyboard).  Now I'm not a fan of some of its choices (it moves open curly onto a new line for/if/while/etc statements, that I would prefer on the same line) and it is over eager to split things onto multiple lines.  But it does help with consistency of formatting.